### DISCUSSION TITLE
What to do in case of api.zonneplan unreachable?

### DISCUSSION THREAD

**Discussion by Paul1865** [fsaris/home-assistant-zonneplan-one#105] [Ideas]
RecentIy had two http response code 500 (service unavailable) when the zonneplan integration was trying to get data from zonneplan.
Result: lots of errors, automations broken etc because the zonneplan current electricity tariff was unavailable.

Is it possible to not touch that sensor (so leave it as it was) when getting a response code >= 400?

I hope that way my derived sensors and automations stay intact. 

Trying in the meantime to create a persistant sensor and restore the zonneplan sensor with the data of the persistant one if it's unavailable. No luck so far :D .

**Comment by Paul1865** [comment]
Best thing I can think of now: a new sensor that only updates when the zonneplan current electricity tariff is available and replace all occurences of the original with this new one...

      - name: "Zonneplan current electricity tariff (last valid)"
        unique_id: zonneplan_current_electricity_tariff_last_valid
        availability: >
          {{ has_value('sensor.zonneplan_current_electricity_tariff') }}
        state: "{{ states('sensor.zonneplan_current_electricity_tariff') }}"
        state_class: measurement
        attributes:
          forecast: >
            {{ state_attr('sensor.zonneplan_current_electricity_tariff', 'forecast') }}
          source: >
            zonneplan_my.yaml
        unit_of_measurement: €
        icon: mdi:currency-eur

**Comment by fsaris** [comment]
Hi,

just ignoring all that's higher than status 400 isn't possible.

A response with `HTTPStatus.UNAUTHORIZED` (401) is handled different as this triggers the re-authentication flow.

If I read the HA code correctly the  current logic of HA is that on a first error the data isn't processed but if the next update attempt also fails the sensors go to unavailable. 

Do you see this behavior often? That the api returns 500 errors?

**Comment by Paul1865** [comment]
No, happened twice and didn't see it any more since then. On Apr 13, 2024 14:27, Frans Saris ***@***.***> wrote:
Hi,
just ignoring all that's higher than status 400 isn't possible.
A response with HTTPStatus.UNAUTHORIZED (401) is handled different as this triggers the re-authentication flow.
If I read the HA code correctly the  current logic of HA is that on a first error the data isn't processed but if the next update attempt also fails the sensors go to unavailable.
Do you see this behavior often? That the api returns 500 errors?
